### PR TITLE
Improve the HoT instances handling

### PR DESCRIPTION
**Pull request by budnix**
### Context

The PR improves the code that is responsible for destroying Handsontable instances used within docs demos. The changes from now on support destroying not only vanilla demos but supports destroying demos that use wrappers as well. 

The registry handler depends on the demo, destroys the Handosntable instance directly, or for frameworks manually trigger the "destroy" lifecycle. So under the hood, all framework-related listeners, components, and other resources can be freed.

### How has this been tested?

https://github.com/handsontable/handsontable/pull/8286#issuecomment-868403905

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #8165
2. #8290
3.

### Affected project(s):
- [ ] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

### CODE REVIEW

**Comment by aninde** [comment]
React page has similar issue, reported here https://github.com/handsontable/handsontable/issues/8289.

**Review by Siemienik — CHANGES_REQUESTED** [review]
@budnix Could you type out "How has this been tested?", please?

I added to `register.js` console logs:
https://github.com/handsontable/handsontable/commit/86c7fa19b50c5abf43b629d87ec40cb5136f3900#

And it does not register instances even for Vue and Vanilla:

https://user-images.githubusercontent.com/2893099/123406631-670d9780-d5ab-11eb-8668-bb7e6f8eb0c4.mp4


https://user-images.githubusercontent.com/2893099/123407300-1b0f2280-d5ac-11eb-9256-342fb7204867.mp4

File: docs/9.0/guides/integrate-with-angular/angular-setting-up-a-locale.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ type Product = {
   selector: 'app-root',
   template: `
     <div>
-      <hot-table [data]="dataset" [colHeaders]="true" [licenseKey]="non-commercial-and-evaluation">
+      <hot-table [data]="dataset" [colHeaders]="true" [licenseKey]="'non-commercial-and-evaluation'">
```
[the remark this answers is not in the document]
Comment (budnix) [reply docs/9.0/guides/integrate-with-angular/angular-setting-up-a-locale.md:39]: That does not work as the name `non-commercial-and-evaluation` then is treated as a prop (that not exist).

[the remark this answers is not in the document]
Comment (Siemienik) [reply docs/9.0/guides/integrate-with-angular/angular-setting-up-a-locale.md:39]: That way I remove `[]` around arg name.

In Angular, you can use args in 3 different ways:
* `[licenseKey]="expression"`
* `licenseKey="string-value"`
* `(click)="onClick($event)"`

For instance `[licenseKey]="1+1"` is same as `licenseKey="2"`

[the remark this answers is not in the document]
Comment (budnix) [reply docs/9.0/guides/integrate-with-angular/angular-setting-up-a-locale.md:39]: Still for that demo this not work :)

[the remark this answers is not in the document]
Comment (budnix) [reply docs/9.0/guides/integrate-with-angular/angular-setting-up-a-locale.md:39]: Ok, I see the problem ... it's about square bracket 👍

**Comment by budnix** [comment]
> @budnix Could you type out "How has this been tested?", please?

Can you re-check by clearing the cache? On my side, everything works. Even with your testing methodology 86c7fa1, I see that everything is registered and then destroyed on the page leaving.

Kapture 2021-06-25 at 12 29 17

About my methodology, I overloaded the Handsontable's`destroy` method and track its callee and track the lifecycles of the framework life-cycles (for Vue `beforeDestroy` etc.).

File: docs/9.0/guides/integrate-with-angular/angular-setting-up-a-locale.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (budnix) [reply docs/9.0/guides/integrate-with-angular/angular-setting-up-a-locale.md]: Fixed within https://github.com/handsontable/handsontable/pull/8286/commits/9f255453965e591f6217c5444e031bcb67cb1148

**Review by Siemienik — APPROVED** [review]
You are right. It works. I had cached pages which were built without `htmlContentRoot`.

I removed `./node_modules/.cache`, and for make sure I run `npm run dosc:start:no-cache`

Everything works ok 🥇

**Pull request merged**